agent/exploits/cve_2024_9634.py

```diff
@@ -0,0 +1,79 @@
+"""Agent Asteroid implementation for CVE-2024-9634"""
+
+import re
+
+import requests
+from packaging import version
+from urllib import parse as urlparse
+
+from agent import definitions
+from agent import exploits_registry
+from agent.exploits import webexploit
+
+VULNERABILITY_TITLE = "PHP Object Injection Vulnerability in GiveWP Plugin"
+VULNERABILITY_REFERENCE = "CVE-2024-9634"
+VULNERABILITY_DESCRIPTION = (
+    "The GiveWP – Donation Plugin and Fundraising Platform plugin for WordPress "
+    "is vulnerable to PHP Object Injection in versions up to 3.16.3. "
+    "This vulnerability allows unauthenticated attackers to inject PHP objects, "
+    "potentially leading to remote code execution through a POP chain."
+)
+RISK_RATING = "CRITICAL"
+MAX_SAFE_VERSION = version.parse("3.16.3")
+
+
+def _is_vulnerable_version(version_str: str) -> bool:
+    """Check if the given version is vulnerable."""
+    return version.parse(version_str.strip()) <= MAX_SAFE_VERSION
+
+
+def _send_request(
+    target: definitions.Target, request: definitions.Request
+) -> requests.Response | None:
+    """Helper to send HTTP requests using the provided session setup."""
+    session = requests.Session()
+    session.max_redirects = webexploit.MAX_REDIRECTS
+    session.verify = False
+
+    try:
+        target_endpoint = urlparse.urljoin(target.origin, request.path)
+        prepared_request = requests.Request(
+            method=request.method, url=target_endpoint, data=request.data
+        ).prepare()
+        if request.headers is not None:
+            request.headers.update(prepared_request.headers)
+            prepared_request.headers = request.headers  # type: ignore
```

Comment:
why this line is not covered with tests?

Reply:
They're covered now.